scene/2d/animated_sprite_2d.cpp

```diff
@@ -154,20 +161,14 @@ void AnimatedSprite2D::_validate_property(PropertyInfo &p_property) const {
 void AnimatedSprite2D::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_INTERNAL_PROCESS: {
-			if (frames.is_null()) {
-				return;
-			}
-			if (!frames->has_animation(animation)) {
-				return;
-			}
-			if (frame < 0) {
+			if (frames.is_null() || !frames->has_animation(animation)) {
 				return;
 			}
 
 			double remaining = get_process_delta_time();
 
 			while (remaining) {
-				double speed = frames->get_animation_speed(animation) * speed_scale;
+				double speed = frames->get_animation_speed(animation) * Math::abs(speed_scale);
 				if (speed == 0) {
 					return; // Do nothing.
 				}
```

Comment:
Sorry, this is also too late, but this should be placed outside the `while`.
```cpp		
double speed = frames->get_animation_speed(animation) * Math::abs(speed_scale);
if (speed == 0) {
	return; // Do nothing.
}
while (remaining) {
```

Reply:
All in a day's work for optimisations.